### ISSUE TITLE
Spatial Audio Issue(s)

### ISSUE THREAD

**Issue by UnknownCoder807** [kingslay/KSPlayer#355]
Am i correct in thinking that 

`KSOptions.isUseAudioRenderer = true `

is the correct way to enable Spatial Audio?

If it is, there is a problem with frames being dropped.

If I play the same video with spatial audio disabled, there are no dropped frames.

Here are the logs when the frames are being dropped

```
KSPlayer: KSPlayerLayer.swift:149 state | playerStateDidChange - buffering
KSPlayer: AVFoundationExtension.swift:134 channelLayout() | KSOptions channelLayout: AudioChannelLayout(mChannelLayoutTag: 7929862, mChannelBitmap: __C.AudioChannelBitmap(rawValue: 0), mNumberChannelDescriptions: 0, mChannelDescriptions: __C.AudioChannelDescription(mChannelLabel: 2863311530, mChannelFlags: __C.AudioChannelFlags(rawValue: 1551364048), mCoordinates: (1e-45, 0.0, 0.0)))
KSPlayer: AVFoundationExtension.swift:160 channelLayout() | out channelLayout mask: 1551 nb_channels: 6
KSPlayer: Resample.swift:190 init(audioDescriptor:outChannel:outSampleRate:) | out channelLayout: AVChannelLayout(order: __C.AVChannelOrder(rawValue: 1), nb_channels: 6, u: __C.AVChannelLayout.__Unnamed_union_u(), opaque: nil)

KSPlayer: MEPlayerItem.swift:706 getVideoOutputRender(force:) | video delay time: 0.6514567070000794 dropped video frame frameCount: 7 frameMaxCount: 8
KSPlayer: MEPlayerItem.swift:706 getVideoOutputRender(force:) | video delay time: 0.5674567070000194 dropped video frame frameCount: 5 frameMaxCount: 8
KSPlayer: MEPlayerItem.swift:706 getVideoOutputRender(force:) | video delay time: 0.4844567069999357 dropped video frame frameCount: 7 frameMaxCount: 8
KSPlayer: MEPlayerItem.swift:706 getVideoOutputRender(force:) | video delay time: 0.4004567069998757 dropped video frame frameCount: 5 frameMaxCount: 8
```

**Comment by kingslay** [comment]
当用isUseAudioRenderer的时候。会有杂音。你那边还会这样吗？

**Comment by kingslay** [comment]
@UnknownCoder807 能否提供drop frame的视频呢？

**Comment by UnknownCoder807** [comment]
> @UnknownCoder807 能否提供drop frame的视频呢？

Yeah, when spatial audio is enabled, the video is dropping frames.  when I turn off spatial audio, the video plays ok.

**Comment by fedalcode** [comment]
I used KSOptions.isUseAudioRenderer = true  with the last swiftUI untouched Ksplayer Demo

1- liveTV (.ts stream) does not work, it  display only the first frame as an image and there is no sound at all

2- after 1-2 minute playback, the sound goes out sync, the sound gets delayed. ıt doesn’t matter if the video is 24 fps or 23.9 fps

Unfortunately there are no logs

**Comment by kingslay** [comment]
可以提供下那个url吗？

**Comment by fedalcode** [comment]
Please check your e-mail

**Comment by UnknownCoder807** [comment]
I am pleased this issue has been brought up again.  I still have issues with video framerate when isUseAudioRenderer is = true.

It's not missing loads of frames, just makes the motion less fluent.  I am seeing this in the Tracy Player and using a 5.1 sound .mp4 file.

There are no logs generated.

**Comment by UnknownCoder807** [comment]
I can also confirm that .ts streams show 1 frame and freeze, no further frames or audio.

Still no logs.

**Comment by UnknownCoder807** [comment]
Recent commit fires the ,ts stream freeze frame issue, but still a problem with missing frames.

**Comment by fedalcode** [comment]
I confirm what @UnknownCoder807 says, the .ts streams plays; there are no freezes, but there is another issue:

when I watch a content in 24 fps or 23.9 fps (the issue is more obvious) after 15 seconds of playback, the sound goes out of sync.

there are no logs, unfortunately.

**Comment by UnknownCoder807** [comment]
I think you should remove the mention of Spatial Audio from the readme as it obviously doesn't work, which is a great shame.

**Comment by fedalcode** [comment]
> I confirm what @UnknownCoder807 says, the .ts streams plays; there are no freezes, but there is another issue:
> 
> when I watch a content in 24 fps or 23.9 fps (the issue is more obvious) after 15 seconds of playback, the sound goes out of sync.
> 
> there are no logs, unfortunately.

I suspect that the audio going out of sync, is also related to this issue too: https://github.com/kingslay/KSPlayer/issues/375

there are still dropped frames for content that have non-integer (23.9 - 59.9 - etc) FPS value.

@kingslay

**Comment by UnknownCoder807** [comment]
Closing as confirmed as it won't work with video, just audio.
